effekt/shared/src/main/scala/effekt/machine/Tree.scala

```diff
@@ -171,11 +169,11 @@ object builtins {
   val Apply: Tag = 0
 
   val Unit: Tag = 0
-  val UnitType = Positive(List(List()))
+  val UnitType = Positive()
 
   val True: Tag = 1
   val False: Tag = 0
-  val BooleanType = Positive(List(List(), List()))
+  val BooleanType = Positive()
```

Comment:
This makes `UnitType` and `BooleanType` indistinguishable, doesn't it? (since `Positive` is a `case class` and thus compared structurally IIRC).

In the JIT backend, I erase `Unit` (and the primitives depend on that), so I kind of need that information.

Reply:
The same holds for `BooleanType` and other data, so I can't implement `Bool` using int registers.